test/kubernetes_unittest.cc

```diff
@@ -0,0 +1,70 @@
+#include "../src/kubernetes.h"
+#include "../src/updater.h"
+#include "../src/configuration.h"
+#include "../src/instance.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class KubernetesTest : public ::testing::Test {
+  protected:
+   MetadataUpdater::ResourceMetadata GetNodeMetadata(
+       const KubernetesReader& reader, const json::Object *node,
+       Timestamp collected_at, bool is_deleted) const
+       throw(json::Exception) {
+      return reader.GetNodeMetadata(node, collected_at, is_deleted);
+   }
+};
+
+TEST_F(KubernetesTest, GetNodeMetadata) {
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: 'TestClusterName'\n"
+    "KubernetesClusterLocation: 'TestClusterLocation'\n"
+    "MetadataIngestionRawContentVersion: 'TestVersion'\n"
+    "InstanceZone: 'TestZone'\n"
+    "InstanceId: 'TestID'\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);
```

Comment:
Ouch, I just realized that we didn't document whether `health_checker` can be a `nullptr`... It makes sense to allow it, but let's also add a guard in the code before every use.
Since this change would be in the code under test, I feel that it's fine to put it in this PR.

Reply:
Added